### PR TITLE
add log ignore for RejectedExecution during upgrade

### PR DESCRIPTION
**Pull request by knifewine**
for cassandra-12364

### CODE REVIEW

**Comment by knifewine** [comment]
@mambocab to review, please.

File: upgrade_tests/upgrade_base.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def __init__(self, *args, **kwargs):
         self.ignore_log_patterns = [
             # Normal occurance. See CASSANDRA-12026. Likely won't be needed after C* 4.0.
             r'Unknown column cdc during deserialization',
+            r'RejectedExecutionException.*ThreadPoolExecutor has shut down',  # see  CASSANDRA-12364
```
Comment (mambocab) [inline upgrade_tests/upgrade_base.py]: I don't know where these exceptions might get raised, so I'm worried that it'd happen in more places than just teardown. Maybe append `.*at org.apache.cassandra.db.ColumnFamilyStore.waitForFlushes`? It sounds like that's the place where this happens.

**Comment by mambocab** [comment]
I couldn't tell from @pcmanus's comment on the Jira ticket whether this was an upgrade-specific error or not. So, this lgtm after my nit, but might not cover all the cases we want it to. @pcmanus, should this error be ignored everywhere, or just in upgrade tests?

File: upgrade_tests/upgrade_base.py (hunk 1 above)
Comment (knifewine) [inline upgrade_tests/upgrade_base.py]: I'm actually unclear as to whether we should ignore any RejectedExecutionException or try to be more specific like you suggested. I wasn't getting this to repro locally so not 100% if the more specific case would have us covered (I also forget whether we can span multiple lines with the ignore regex, if that will work). If we are in fact being too general then the same pattern in upgrade_through_versions_test.py will need a fixup too.

Comment (knifewine) [inline upgrade_tests/upgrade_base.py]: note on the u_t_v tests: looks like CASSANDRA-7727 was what prompted a similar pattern to be ignored in those tests. Looks to be a RejectedExecutionException but with a different backtrace than the one noted on 12364.

**Comment by ptnapoleon** [comment]
Bump, @mambocab. Since Sylvain is out, could you investigate with someone else?

**Comment by knifewine** [comment]
We're still seeing failures on CI due to not ignoring logs properly. 

http://cassci.datastax.com/view/cassandra-3.9/job/cassandra-3.9_dtest_upgrade/lastCompletedBuild/testReport/upgrade_tests.paging_test/TestPagingWithDeletionsNodes3RF3_Upgrade_current_3_x_To_indev_3_x/test_single_cell_deletions/

Unfortunately, this failure would not be caught by the more specific pattern suggested above, and I still think the more generous regex in my pr would be fine here. At the very least maybe we can merge in this version and spin off another issue to investigate making things more specific at a later time.

Getting this in would help get the CI failure baseline down even more and get rid of these as lingering issues (which are really just noise).

**Comment by knifewine** [comment]
@ptnapoleon @mambocab bump. thoughts here?

**Comment by pcmanus** [comment]
It's not an upgrade specific thing and could be triggered anytime a node is shutdown (gently). Basically, it's a race in C* gentle shutdown with no real consequence other that log pollution. It's still pollution and we should fix it where it happens, and we have already done so in other places in the past, but we never spend the cycles finding all the places where this can happen and fixing them. And for the reasons above, it's not high on our priority list. Hence the suggestion of ignoring for test purposes.

As for when to ignore, ideally we should ignore it every time we shut down a node gently, but as I imagine it might be a pain to do that exactly, I don't think it's we're taking much risk in just ignoring it all the time (note that I'm mostly saying to ignore in term of failing the whole test, we could still log some form of warning so we have some reminder that this is not ideal).

**Comment by ptnapoleon** [comment]
I'm +1. I think the risk of this covering up a different error is small enough.

**Pull request merged**